### ISSUE TITLE
Hotkey assignments are lost when moving between servers.

### ISSUE THREAD

**Issue by WCPistolPete** [SmartlyDressedGames/Unturned-3.x-Community#2186]
Log in to server A. Immediately go to Inventory screen, assign hotkeys. Log out. Go to server B. Immediately go to Inventory screen, assign hotkeys. Go back to server A. Previous hotkeys no longer assigned. Note that I tested by simply assigning hotkeys. No text chat was involved.

**Comment by SDGNelson** [comment]
Hmm thanks for the report, will check

**Comment by SDGNelson** [comment]
Double-checked and it seemed to work normally. In case it helps narrow it down: the hotkeys are saved per IPv4/port combination, so was the server restarted between connections on a different address? The hotkey files are in Unturned/Worlds/Hotkeys if you want to check that they are being written

**Comment by WCPistolPete** [comment]
Just check my client. Even though I just assigned hotkeys again today going between servers, the last file date for hotkey assignments on my computer was 19 Nov.

**Comment by SDGNelson** [comment]
Interesting that no files have been written since then. After it happens again could you attach your Client.log file please? I wonder if there is a write permissions exception

**Comment by WCPistolPete** [comment]
[Client.log](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/files/5681150/Client.log)

**Comment by SDGNelson** [comment]
Thanks for the log. Unfortunately no errors. Is anyone else having this problem? I will add some logging for the next update to see what the game is doing

**Comment by WCPistolPete** [comment]
I use the same setup across all servers: Upper left is Bandage, #3, to the right of that is a claim flag, #4 and to the right of that is umbrella #5. Haven't been able to spot a pattern with the loss of hotkey assignments. Just went into PvE1 and no hotkeys at all. PvE2, lost the bind for the claim flag. PvE3, all good. PvE4, lost the bind for the umbrella. PvE5m all good. PvE6, well, blackscreened.  This is the log from the latest 1-5/crash hop where I just assigned hotkeys.

[Client.log](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/files/5681230/Client.log)

When I went to join PvE6, blackscreened. Everything was still running in the process list. That goes back to the previously reported crashes. The uGUI update does something with 2D that older video cards/drivers just don't like. That's the flickering, artifact and protected memory errors. I'll send that stuff separately.
